justlevine/woocommerce
# Preserve input classes when updating state field

Described in
https://github.com/woocommerce/woocommerce/issues/23020

### All Submissions:


### Changes proposed in this Pull Request:

Added `var input_classes` that saves any input classes added to the state field.
Adds `input_classes` to the class-list whenever the state field is `replaceWith()`ed.

Note: it is not added to the hidden state (for css compatibility reasons)

Closes #23020 .

### How to test the changes in this Pull Request:

1. Add a custom `input_class` to the state field via the `woocommerce_form_field_args` filter.
2. Either set `SCRIPT_DEBUG` to true, or generate a minified version country-select.js
3. Visit Checkout, Account Details, or anywhere else your fields are output, choose a country, and inspect to verify your input classes are included in the state field.

### Other information:
Ideally, other field attributes one can manually add to the state field would be preserved as well, but i'm not familiar enough with all the possibilities to account for them all. The method for including them would be the same: save the initial attributes to a variable, and use them in the replaceWith().





### Changelog entry

> Preserve the State field's custom classes when selecting an option from the Country dropdown.
